core/src/test/java/org/elasticsearch/index/query/MultiMatchQueryBuilderTests.java

```diff
@@ -0,0 +1,185 @@
+import org.apache.lucene.search.DisjunctionMaxQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.elasticsearch.index.search.MatchQuery;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.index.query.QueryBuilders.multiMatchQuery;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertBooleanSubQuery;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+public class MultiMatchQueryBuilderTests extends AbstractQueryTestCase<MultiMatchQueryBuilder> {
+
+    @Override
+    protected MultiMatchQueryBuilder doCreateTestQueryBuilder() {
+        String fieldName = randomFrom(STRING_FIELD_NAME, INT_FIELD_NAME, DOUBLE_FIELD_NAME, BOOLEAN_FIELD_NAME, DATE_FIELD_NAME);
+        MultiMatchQueryBuilder query = new MultiMatchQueryBuilder(getRandomValueForFieldName(fieldName), fieldName);
+        // field with random boost
+        if (randomBoolean()) {
+            query.field(fieldName, randomFloat() * 10);
+        }
+        // field with a pattern
+        if (randomBoolean()) {
+            query.field('*' + fieldName);
+        }
+        // sets other parameters of the multi match query
+        if (randomBoolean()) {
+            query.type(randomFrom(MultiMatchQueryBuilder.Type.values()));
+        }
+        if (randomBoolean()) {
+            query.operator(randomFrom(Operator.values()));
+        }
+        if (randomBoolean()) {
+            query.analyzer(randomFrom("simple", "keyword", "whitespace"));
+        }
+        if (randomBoolean()) {
+            query.slop(randomIntBetween(0, 5));
+        }
+        if (randomBoolean()) {
+            query.fuzziness(randomFuzziness(fieldName));
+        }
+        if (randomBoolean()) {
+            query.prefixLength(randomIntBetween(0, 5));
+        }
+        if (randomBoolean()) {
+            query.maxExpansions(randomIntBetween(0, 5));
+        }
+        if (randomBoolean()) {
+            query.minimumShouldMatch(randomMinimumShouldMatch());
+        }
+        if (randomBoolean()) {
+            query.fuzzyRewrite(getRandomRewriteMethod());
+        }
+        if (randomBoolean()) {
+            query.useDisMax(randomBoolean());
+        }
+        if (randomBoolean()) {
+            query.tieBreaker(randomFloat() * 10);
+        }
+        if (randomBoolean()) {
+            query.lenient(randomBoolean());
+        }
+        if (randomBoolean()) {
+            query.cutoffFrequency((float) 10 / randomIntBetween(1, 100));
+        }
+        if (randomBoolean()) {
+            query.zeroTermsQuery(randomFrom(MatchQuery.ZeroTermsQuery.values()));
+        }
+        return query;
+    }
+
+    @Override
+    protected void doAssertLuceneQuery(MultiMatchQueryBuilder queryBuilder, Query query, QueryShardContext context) throws IOException {
+
```

Comment:
Maybe put some comment why we don't to further checks here or if there are more specialized test below? Otherwise one might wonder why this is here.